### ISSUE TITLE
Custom primary key lost on reload

### ISSUE THREAD

**Issue by atomical** [pat/thinking-sphinx#1125]
I'm not sure if this is a Rails (4.2.8) issue or a Thinking Sphinx (4.0.0) issue. In the development environment after a reload the the primary_key on the model doesn't appear to be set and calling count on an instance of ThinkingSphinx::Search tries to do the query with _id_ instead of the primary key.

Edit: After doing some digging this appears to be happening with all models, but the default is :id so it doesn't error out for those models.

```
ThinkingSphinx::Middlewares::ActiveRecordTranslator
(byebug) index_for(model)
#<struct primary_key=:id>
```

**Comment by pat** [comment]
I'm surprised this issue hasn't cropped up before, but thanks for logging it! I've pushed a fix to the develop branch, and will be a part of the next release (hopefully within the coming week). If you want something in your app now, you can put the following in your `config/environment.rb` file:

```ruby
config.to_prepare do
  ThinkingSphinx::Configuration.reset
end
```

**Comment by atomical** [comment]
Thanks, we appreciate it!

**Comment by pat** [comment]
Just published v4.2.0 which includes this PR :)
